src/server/routes/collection.js

```diff
@@ -85,4 +85,31 @@ router.get('/:collectionId', (req, res) => {
 	res.status(200).send(collection);
 });
 
+/* eslint-disable no-await-in-loop */
+router.post('/:collectionId/add', auth.isAuthenticated, auth.isCollectionOwnerOrCollaborator, async (req, res) => {
+	const {bbids} = req.body;
+	const {collection} = res.locals;
+	try {
+		const {UserCollectionItem} = req.app.locals.orm;
+		for (const bbid of bbids) {
+			try {
+				await new UserCollectionItem({
+					bbid,
+					collectionId: collection.id
+				}).save(null, {method: 'insert'});
+			}
+			catch (err) {
+				if (err.constraint !== 'user_collection_item_pkey') {
```

Comment:
I wish there was a better way to check if it's an "entry already exists" error, but nothing I tried is better than this. So be it.
A comment would be useful above this line to explain why we're doing this, as it's not clear unless you're very familiar with the db errors (which I'll admit I'm not).

However in this block you would `throw err` instead of logging it so that it can be dealt with by the parent catch block. Otherwise you're in a situation where the addition failed but you're sending a success response.